client/app/(todos)/components/TaskItem.tsx

```diff
@@ -0,0 +1,26 @@
+import { Checkbox } from '@/_components/ui'
+import { TaskType } from '@/_types'
+import { useUpdateStatusMutation } from '../hooks'
+
+const TaskItem = ({ task }: { task: TaskType }) => {
+  const { mutate: updateStatus } = useUpdateStatusMutation()
+
+  const handleUpdateTaskStatus = () => {
+    updateStatus({ taskId: task.id, isCompleted: !task.isCompleted })
+  }
+
+  return (
+    <li className='flex items-center justify-between gap-6 border-b-[1px] border-slate-400 pb-1'>
+      <p className='max-w-64 truncate text-lg font-semibold md:max-w-36 lg:max-w-72'>
+        {task.title}
+      </p>
+      <Checkbox
+        className='h-6 w-6 md:h-8 md:w-8'
+        checked={task.isCompleted}
+        onClick={handleUpdateTaskStatus}
+      />
```

Comment:
Can you add disabled while the mutation is running? You don't want a user to push this button many times before it finishes

Reply:
Fixed it!
https://github.com/HAK2024/fairy-share/pull/71/commits/7821e941b5e5a8dd81164ab6149b427502f0eb01